src/ctap2/attestation.rs

```diff
@@ -270,23 +220,39 @@ impl<'de> Deserialize<'de> for AuthenticatorData {
                 formatter.write_str("a byte array")
             }
 
-            fn visit_bytes<E>(self, v: &[u8]) -> Result<Self::Value, E>
+            fn visit_bytes<E>(self, input: &[u8]) -> Result<Self::Value, E>
             where
                 E: SerdeError,
             {
-                parse_ad(v)
-                    .map(|(_input, value)| value)
-                    .map_err(|e| match e {
-                        NomErr::Incomplete(nom::Needed::Size(len)) => {
-                            E::invalid_length(v.len(), &format!("{}", v.len() + len.get()).as_ref())
-                        }
-                        NomErr::Incomplete(nom::Needed::Unknown) => {
-                            E::invalid_length(v.len(), &"unknown") // We don't know the expected value
-                        }
-                        // TODO(baloo): is that enough? should we be more
-                        //              specific on the error type?
-                        e => E::custom(e.to_string()),
-                    })
+                let mut cursor = Cursor::new(input);
+                let mut rp_id_hash_raw = [0u8; 32];
+                cursor
+                    .read_exact(&mut rp_id_hash_raw)
+                    .map_err(|_| serde_parse_err("32 bytes"))?;
+                let rp_id_hash = RpIdHash(rp_id_hash_raw);
+
+                // be conservative, there is a couple of reserved values in
+                // AuthenticatorDataFlags, just truncate the one we don't know
```

Comment:
The previous comment was correct. `AuthenticatorDataFlags` includes all of the reserved flags, so `from_bits_truncate` is lossless. We use `from_bits_truncate` because of its return value. See #263.

Reply:
Oh, seems like the new comment got lost in a rebase.